modules/Microsoft.CDN/profiles/endpoints/deploy.bicep

```diff
@@ -32,36 +32,41 @@ resource defaultTelemetry 'Microsoft.Resources/deployments@2021-04-01' = if (ena
   }
 }
 
-resource profileName_endpoint 'microsoft.cdn/profiles/endpoints@2021-06-01' = if (!empty(endpointName)){
+resource endpoint 'microsoft.cdn/profiles/endpoints@2021-06-01' = {
   parent: profile
-  name: endpointName
+  name: name
   location: location
   properties: endpointProperties
   tags: tags
 }
 
-resource profile_EndpointOrigins 'Microsoft.Cdn/profiles/endpoints/origins@2021-06-01' = if (!empty(endpointProperties.origins)){
-  parent: profileName_endpoint
-  name: endpointProperties.origins[0].name
-  properties: {
-    hostName: endpointProperties.origins[0].properties.hostName
-    httpPort: endpointProperties.origins[0].properties.httpPort
-    enabled: endpointProperties.origins[0].properties.enabled
-    httpsPort: endpointProperties.origins[0].properties.httpsPort
+module endpoint_origins 'origins/deploy.bicep' = [for origin in endpointProperties.origins: {
+  name: '${name}-origins-${origin.name}'
+  params: {
+    profileName: profile.name
+    endpointName: name
+    name: origin.name
+    hostName: origin.properties.hostName
+    httpPort: contains(origin.properties, 'httpPort') ? origin.properties.httpPort : null
```

Comment:
This is supprising. Since when can we pass down `null`? Is this a new feature and does it mean it will automatically use the default of the target module?

Reply:
We just established the answer is sadly no